src/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -990,9 +990,11 @@ public void SetCookies(Uri uri, string cookieHeader)
         }
     }
 
+    // PathList needs to be public in order to maintain binary serialization compatibility as the System shim
+    // needs to have access to type-forward it.
     [Serializable]
     [System.Runtime.CompilerServices.TypeForwardedFrom("System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
-    internal struct PathList
+    public struct PathList
```

Comment:
> So we are safe to make fundamental changes in the future as long as serialization compatibility is guaranteed.

I hope so. I just worry that changes to public members won't be allowed in the future out of an abundance of caution because it's now a public type in the implementation, even if it isn't publicly documented/exposed in the ref assembly.

I wonder if it'd be prudent to make it a **sealed** class with all members internal now.

At the very least, maybe the comment should mention that `PathList` is not intended to be used directly by external callers and should not be exposed in the reference assembly. It's an implementation detail and its public surface area may be modified in the future.